libs/components/src/next/form/radio/vl-radio-group.component.ts

```diff
@@ -0,0 +1,224 @@
+            } else {
+                radio.removeAttribute('checked');
+            }
+        });
+    }
+
+    private resetAllRadios(): void {
+        this.getRadios()?.forEach((radio) => {
+            radio.removeAttribute('checked');
+        });
+    }
+
+    private preventDefaultAndFocus = (event: Event) => {
+        const targetElement = event.target as VlRadioComponent;
+        targetElement?.validationTarget?.focus();
+        event.preventDefault();
+    };
+
+    private preventDefaultEventOnAllInputsOfAllRadio(readonly: boolean): void {
+        this.getRadios()?.forEach((radio) => {
+            if (readonly) {
+                radio.addEventListener('click', this.preventDefaultAndFocus);
+            } else {
+                radio.removeEventListener('click', this.preventDefaultAndFocus);
+            }
+        });
+    }
+
+    private setAttributeForCondition = (radio: VlRadioComponent, attribute: string, condition: boolean) => {
+        if (condition) {
+            radio.setAttribute(attribute, '');
+        } else {
+            radio.removeAttribute(attribute);
+        }
+    };
+
+    private updateRadiosForAttribute(attribute: string): void {
+        const attributeValue = attribute as unknown as keyof VlRadioGroupComponent;
+        this.getRadios()?.forEach((radio) => this.setAttributeForCondition(radio, attribute, this[attributeValue]));
+    }
+
+    private getRadios(): VlRadioComponent[] {
+        const allRadios = Array.from(this.querySelectorAll<VlRadioComponent>('vl-radio-next'));
+        return allRadios;
+    }
+
+    private checkSelectedRadio(selectedRadio: VlRadioComponent) {
+        this.getRadios()?.forEach((radio) => {
+            const value = radio.getAttribute('value');
+            if (radio !== selectedRadio) {
+                radio.removeAttribute('checked');
+            } else {
+                if (!radio.hasAttribute('checked')) radio.setAttribute('checked', '');
+                this.value = value ? value : '';
+                this.setValue(value);
+            }
+        });
+    }
+
+    private handleKeyDown = (event: KeyboardEvent) => {
+        const arrows = ['ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown'];
+        if (arrows.includes(event.code)) {
+            event.preventDefault(); // Voorkom dat de pagina scrollt
+            this.navigateRadioButtons(event.code);
+        }
+        if (event.code === 'Space') {
+            const focusedRadio = this.getRadios()?.find((radio) => radio === document.activeElement);
+            if (focusedRadio && !focusedRadio.hasAttribute('checked')) this.checkSelectedRadio(focusedRadio);
+        }
+    };
+
+    private navigateRadioButtons(direction: string) {
+        const radios = this.getRadios();
+        const currentIndex = radios.findIndex((radio) => radio.hasAttribute('checked'));
+        let newIndex;
+
+        if (direction === 'ArrowLeft' || direction === 'ArrowUp') {
```

Comment:
Moet dit met de arrows kunnen? Is navigeren met de tab toets niet voldoende?

Reply:
Dit kon vroeger ook en dit is ook native gedrag daarom heb ik diezelfde functionaliteit behouden.